### ISSUE TITLE
Plugins don't work

### ISSUE THREAD

**Issue by ilijabojanovic** [TykTechnologies/tyk#2072]
**Do you want to request a *feature* or report a *bug*?**
BUG
**What is the current behavior?**
When we enable rewrite and virtual plugins, they don't work
**What is the expected behavior?**
All plugins should work
**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem**
1. Start on-prem env
2. Import api definition from task
3. Trigger `GET` call to {dashboard_url}/test/ilija1

Actual:
Nothing happens

Expected:
Virtual endpoint should be executed

Testing definition:
```
{
  "api_definition": {
    "api_id": "20411c5ea7c748e174eb5fd8be404b48",
    "upstream_certificates": {},
    "use_keyless": false,
    "enable_coprocess_auth": false,
    "custom_middleware": {
      "pre": [],
      "post": [],
      "post_key_auth": [],
      "auth_check": {
        "name": "",
        "path": "",
        "require_session": false
      },
      "response": [],
      "driver": "",
      "id_extractor": {
        "extract_from": "",
        "extract_with": "",
        "extractor_config": {}
      }
    },
    "disable_quota": false,
    "custom_middleware_bundle": "",
    "cache_options": {
      "enable_cache": true,
      "enable_upstream_cache_control": false,
      "cache_timeout": 60,
      "cache_response_codes": [],
      "cache_all_safe_requests": false
    },
    "enable_ip_blacklisting": false,
    "tag_headers": [],
    "pinned_public_keys": {},
    "domain": "",
    "openid_options": {
      "providers": [],
      "segregate_by_client": false
    },
    "active": true,
    "config_data": {
      "foo": "bar"
    },
    "notifications": {
      "oauth_on_keychange_url": "",
      "shared_secret": ""
    },
    "auth": {
      "auth_header_name": "Authorization",
      "use_param": false,
      "param_name": "",
      "use_cookie": false,
      "use_certificate": false,
      "cookie_name": ""
    },
    "check_host_against_uptime_tests": false,
    "blacklisted_ips": [],
    "hmac_allowed_clock_skew": -1,
    "uptime_tests": {
      "check_list": [],
      "config": {
        "service_discovery": {
          "use_discovery_service": false,
          "query_endpoint": "",
          "use_nested_query": false,
          "parent_data_path": "",
          "data_path": "",
          "cache_timeout": 60
        }
      }
    },
    "enable_jwt": false,
    "name": "test",
    "slug": "test",
    "oauth_meta": {
      "allowed_access_types": [],
      "allowed_authorize_types": [],
      "auth_login_redirect": ""
    },
    "CORS": {
      "enable": false,
      "max_age": 24,
      "allow_credentials": false,
      "exposed_headers": [],
      "allowed_headers": [],
      "options_passthrough": false,
      "debug": false,
      "allowed_origins": [],
      "allowed_methods": []
    },
    "event_handlers": {
      "events": {}
    },
    "proxy": {
      "target_url": "http://httpbin.org/",
      "service_discovery": {
        "cache_timeout": 0,
        "parent_data_path": "",
        "query_endpoint": "",
        "use_discovery_service": false,
        "_sd_show_port_path": false,
        "target_path": "",
        "use_nested_query": false,
        "data_path": "",
        "port_data_path": ""
      },
      "check_host_against_uptime_tests": false,
      "transport": {
        "ssl_ciphers": [],
        "ssl_min_version": 0,
        "proxy_url": ""
      },
      "target_list": [],
      "preserve_host_header": false,
      "strip_listen_path": true,
      "enable_load_balancing": false,
      "listen_path": "/test/"
    },
    "client_certificates": [],
    "use_basic_auth": false,
    "version_data": {
      "not_versioned": true,
      "versions": {
        "Default": {
          "name": "Default",
          "expires": "",
          "override_target": "",
          "extended_paths": {
            "validate_json": [
              {
                "method": "POST",
                "path": "/post",
                "schema": {
                  "type": "object",
                  "required": [
                    "name",
                    "email"
                  ],
                  "properties": {
                    "id": {
                      "type": "number"
                    },
                    "name": {
                      "type": "string"
                    },
                    "email": {
                      "type": "string"
                    },
                    "active": {
                      "type": "boolean"
                    },
                    "something": {
                      "type": "array",
                      "items": {
                        "type": "string"
                      }
                    }
                  }
                },
                "error_response_code": 400
              }
            ],
            "url_rewrites": [
              {
                "method": "GET",
                "timeout": 0,
                "path": "/rewrite1",
                "add_headers": {},
                "samples": 100,
                "rewrite_to": "",
                "template_data": {
                  "input_type": "",
                  "template_mode": "",
                  "enable_session": false,
                  "template_source": ""
                },
                "function_source_uri": "",
                "threshold_percent": 0.1,
                "delete_headers": [],
                "response_function_name": "myVirtualHandler",
                "_lists": [],
                "match_pattern": "/rewrite1",
                "function_source_type": "blob",
                "triggers": [
                  {
                    "on": "all",
                    "options": {
                      "query_val_matches": {
                        "show_env": {
                          "match_rx": "1"
                        }
                      },
                      "header_matches": {},
                      "path_part_matches": {},
                      "session_meta_matches": {},
                      "payload_matches": {
                        "match_rx": ""
                      }
                    },
                    "rewrite_to": "/get?show_env=2"
                  }
                ],
                "return_to_service_after": 60,
                "size_limit": 0,
                "use_session": false,
                "method_actions": {}
              },
              {
                "method": "GET",
                "timeout": 0,
                "path": "/rewrite",
                "add_headers": {},
                "samples": 100,
                "rewrite_to": "/headers",
                "template_data": {
                  "input_type": "",
                  "template_mode": "",
                  "enable_session": false,
                  "template_source": ""
                },
                "function_source_uri": "",
                "threshold_percent": 0.1,
                "delete_headers": [],
                "response_function_name": "myVirtualHandler",
                "_lists": [],
                "match_pattern": "/rewrite",
                "function_source_type": "blob",
                "triggers": [],
                "return_to_service_after": 60,
                "size_limit": 0,
                "use_session": false,
                "method_actions": {}
              }
            ],
            "transform_headers": [
              {
                "method": "GET",
                "timeout": 0,
                "path": "/headers",
                "add_headers": {
                  "request": "add"
                },
                "samples": 100,
                "rewrite_to": "",
                "template_data": {
                  "input_type": "",
                  "template_mode": "",
                  "enable_session": false,
                  "template_source": ""
                },
                "function_source_uri": "",
                "threshold_percent": 0.1,
                "delete_headers": [],
                "response_function_name": "myVirtualHandler",
                "_lists": [],
                "match_pattern": "",
                "function_source_type": "blob",
                "return_to_service_after": 60,
                "size_limit": 0,
                "use_session": false,
                "method_actions": {}
              }
            ],
            "transform_response_headers": [
              {
                "method": "GET",
                "timeout": 0,
                "path": "/headers",
                "add_headers": {
                  "response": "add"
                },
                "samples": 100,
                "rewrite_to": "",
                "template_data": {
                  "input_type": "",
                  "template_mode": "",
                  "enable_session": false,
                  "template_source": ""
                },
                "function_source_uri": "",
                "threshold_percent": 0.1,
                "delete_headers": [],
                "response_function_name": "myVirtualHandler",
                "_lists": [],
                "match_pattern": "",
                "function_source_type": "blob",
                "return_to_service_after": 60,
                "size_limit": 0,
                "use_session": false,
                "method_actions": {}
              }
            ],
            "virtual": [
              {
                "method": "POST",
                "timeout": 0,
                "path": "/ilija2",
                "add_headers": {},
                "samples": 100,
                "rewrite_to": "",
                "template_data": {
                  "input_type": "",
                  "template_mode": "",
                  "enable_session": false,
                  "template_source": ""
                },
                "function_source_uri": "ZnVuY3Rpb24gdGVzdFZpcnREYXRhKHJlcXVlc3QsIHNlc3Npb24sIGNvbmZpZykgewogICAgdmFyIHJlc3AgPSB7CiAgICAgICAgQm9keTogcmVxdWVzdC5Cb2R5ICsgIiBhZGRlZCBib2R5IiwgCiAgICAgICAgSGVhZGVyczogewogICAgICAgICAgICAiZGF0YS1mb28iOiBjb25maWcuY29uZmlnX2RhdGEuZm9vCiAgICAgICAgfSwKICAgICAgICBDb2RlOiAyMDIKICAgIH0KICAgIHJldHVybiBUeWtKc1Jlc3BvbnNlKHJlc3AsIHNlc3Npb24ubWV0YV9kYXRhKSAgIAp9",
                "threshold_percent": 0.1,
                "delete_headers": [],
                "response_function_name": "testVirtData",
                "_lists": [],
                "match_pattern": "",
                "function_source_type": "blob",
                "return_to_service_after": 60,
                "size_limit": 0,
                "use_session": false,
                "method_actions": {}
              },
              {
                "method": "GET",
                "timeout": 0,
                "path": "/ilija",
                "add_headers": {},
                "samples": 100,
                "rewrite_to": "",
                "template_data": {
                  "input_type": "",
                  "template_mode": "",
                  "enable_session": false,
                  "template_source": ""
                },
                "function_source_uri": "ZnVuY3Rpb24gYmF0Y2hUZXN0IChyZXF1ZXN0LCBzZXNzaW9uLCBjb25maWcpIHsKICAgICAgICAvLyBTZXQgdXAgYSByZXNwb25zZSBvYmplY3QKICAgICAgICB2YXIgcmVzcG9uc2UgPSB7CiAgICAgICAgICAgIEJvZHk6ICIiCiAgICAgICAgICAgIEhlYWRlcnM6IHsKICAgICAgICAgICAgICAgICJ0ZXN0IjogInZpcnR1YWwtaGVhZGVyLTEiLAogICAgICAgICAgICAgICAgInRlc3QtMiI6ICJ2aXJ0dWFsLWhlYWRlci0yIiwKICAgICAgICAgICAgICAgICJjb250ZW50LXR5cGUiOiAiYXBwbGljYXRpb24vanNvbiIKICAgICAgICAgICAgfSwKICAgICAgICAgICAgQ29kZTogMjAwCiAgICAgICAgfQogICAgCiAgICAgICAgLy8gQmF0Y2ggcmVxdWVzdAogICAgICAgIHZhciBiYXRjaCA9IHsKICAgICAgICAgICAgInJlcXVlc3RzIjogWwogICAgICAgICAgICAgICAgewogICAgICAgICAgICAgICAgICAgICJtZXRob2QiOiAiR0VUIiwKICAgICAgICAgICAgICAgICAgICAiaGVhZGVycyI6IHsKICAgICAgICAgICAgICAgICAgICAgICAgIngtdHlrLXRlc3QiOiAiMSIsCiAgICAgICAgICAgICAgICAgICAgICAgICJ4LXR5ay12ZXJzaW9uIjogIjEuMiIsCiAgICAgICAgICAgICAgICAgICAgICAgICJhdXRob3JpemF0aW9uIjogIjFkYmM4M2I5YzQzMTY0OWQ3Njk4ZmFhOTc5N2UyOTAwZiIKICAgICAgICAgICAgICAgICAgICB9LAogICAgICAgICAgICAgICAgICAgICJib2R5IjogIiIsCiAgICAgICAgICAgICAgICAgICAgInJlbGF0aXZlX3VybCI6ICJodHRwOi8vaHR0cGJpbi5vcmcvaGVhZGVycyIKICAgICAgICAgICAgICAgIH0sCiAgICAgICAgICAgICAgICB7CiAgICAgICAgICAgICAgICAgICAgIm1ldGhvZCI6ICJHRVQiLAogICAgICAgICAgICAgICAgICAgICJoZWFkZXJzIjoge30sCiAgICAgICAgICAgICAgICAgICAgImJvZHkiOiAiIiwKICAgICAgICAgICAgICAgICAgICAicmVsYXRpdmVfdXJsIjogImh0dHA6Ly9odHRwYmluLm9yZy91c2VyLWFnZW50IgogICAgICAgICAgICAgICAgfQogICAgICAgICAgICBdLAogICAgICAgICAgICAic3VwcHJlc3NfcGFyYWxsZWxfZXhlY3V0aW9uIjogZmFsc2UKICAgICAgICB9CiAgICAKICAgICAgICBsb2coIltWaXJ0dWFsIFRlc3RdIE1ha2luZyBVcHN0cmVhbSBCYXRjaCBSZXF1ZXN0IikKICAgICAgICB2YXIgbmV3Qm9keSA9IFR5a0JhdGNoUmVxdWVzdChKU09OLnN0cmluZ2lmeShiYXRjaCkpCiAgICAKICAgICAgICAvLyBXZSBrbm93IHRoYXQgdGhlIHJlcXVlc3RzIHJldHVybiBKU09OIGluIHRoZWlyIGJvZHksIGxldHMgZmxhdHRlbiBpdAogICAgICAgIHZhciBhc0pTID0gSlNPTi5wYXJzZShuZXdCb2R5KQogICAgICAgIGZvciAodmFyIGkgaW4gYXNKUykgewogICAgICAgICAgICBhc0pTW2ldLmJvZHkgPSBKU09OLnBhcnNlKGFzSlNbaV0uYm9keSkKICAgICAgICB9CiAgICAKICAgICAgICAvLyBXZSBuZWVkIHRvIHNlbmQgYSBzdHJpbmcgb2JqZWN0IGJhY2sgdG8gVHlrIHRvIGVtYmVkIGluIHRoZSByZXNwb25zZQogICAgICAgIHJlc3BvbnNlLkJvZHkgPSBKU09OLnN0cmluZ2lmeShhc0pTKQogICAgCiAgICAgICAgcmV0dXJuIFR5a0pzUmVzcG9uc2UocmVzcG9uc2UsIHNlc3Npb24ubWV0YV9kYXRhKQogICAgCiAgICB9CiAgICBsb2coIkJhdGNoIFRlc3QgaW5pdGlhbGlzZWQiKQ==",
                "threshold_percent": 0.1,
                "delete_headers": [],
                "response_function_name": "batchTest",
                "_lists": [],
                "match_pattern": "",
                "function_source_type": "blob",
                "return_to_service_after": 60,
                "size_limit": 0,
                "use_session": false,
                "method_actions": {}
              }
            ],
            "ignored": [
              {
                "method": "",
                "timeout": 0,
                "path": "/mock",
                "add_headers": {},
                "samples": 100,
                "rewrite_to": "",
                "template_data": {
                  "input_type": "",
                  "template_mode": "",
                  "enable_session": false,
                  "template_source": ""
                },
                "function_source_uri": "",
                "threshold_percent": 0.1,
                "delete_headers": [],
                "response_function_name": "myVirtualHandler",
                "_lists": [],
                "match_pattern": "",
                "function_source_type": "blob",
                "return_to_service_after": 60,
                "size_limit": 0,
                "use_session": false,
                "method_actions": {
                  "GET": {
                    "action": "reply",
                    "code": 200,
                    "data": "{\n    \"mock\":\"response\"\n    }",
                    "headers": {
                      "mock": "header"
                    }
                  }
                }
              },
              {
                "method": "",
                "timeout": 0,
                "path": "/ip",
                "add_headers": {},
                "samples": 100,
                "rewrite_to": "",
                "template_data": {
                  "input_type": "",
                  "template_mode": "",
                  "enable_session": false,
                  "template_source": ""
                },
                "function_source_uri": "",
                "threshold_percent": 0.1,
                "delete_headers": [],
                "response_function_name": "myVirtualHandler",
                "_lists": [],
                "match_pattern": "",
                "function_source_type": "blob",
                "return_to_service_after": 60,
                "size_limit": 0,
                "use_session": false,
                "method_actions": {
                  "GET": {
                    "action": "no_action",
                    "code": 200,
                    "headers": {}
                  }
                }
              }
            ]
          },
          "global_headers": {
            "add_header": "qa"
          },
          "global_headers_remove": [],
          "global_size_limit": 0,
          "use_extended_paths": true
        }
      },
      "default_version": ""
    },
    "use_standard_auth": true,
    "disable_rate_limit": false,
    "definition": {
      "key": "x-api-version",
      "location": "header"
    },
    "use_oauth2": false,
    "allowed_ips": [],
    "org_id": "5ae9bb624c802a00106be456",
    "enable_ip_whitelisting": false,
    "global_rate_limit": {
      "rate": 0,
      "per": 0
    },
    "enable_context_vars": true,
    "tags": [],
    "strip_auth_data": true,
    "id": "5ae9bbb14c802a00106be45a",
    "enable_signature_checking": false,
    "use_openid": false,
    "enable_batch_request_support": true,
    "response_processors": [
      {
        "name": "header_injector",
        "options": {}
      }
    ],
    "use_mutual_tls_auth": false,
    "auth_provider": {
      "name": "",
      "storage_engine": "",
      "meta": {}
    },
    "do_not_track": false,
    "dont_set_quota_on_create": false,
    "expire_analytics_after": 0,
    "session_lifetime": 0
  },
  "hook_references": [],
  "is_site": false,
  "sort_by": 0
}
```
**Which versions of Tyk affected by this issue? Did this work in previous versions of Tyk?**
Release-2.7, Release-2.8 and master

**Comment by buger** [comment]
https://github.com/TykTechnologies/tyk/pull/2073 was merged to `release-2.8` branch
<details><summary></summary>Created via API</details>

**Comment by buger** [comment]
https://github.com/TykTechnologies/tyk/pull/2073 was merged to `release-2.7` branch
<details><summary></summary>Created via API</details>

**Comment by ilijabojanovic** [comment]
Verified
